### DISCUSSION TITLE
How to output logs from an init container to the UI?

### DISCUSSION THREAD

**Discussion by doublefelix7** [apache/airflow#26372] [Q&A] [answered]
In my Airflow environment, I've got a pod that launches whenever a DAG starts; the pod consists of an init-container with git-sync and then the base container with the main Airflow DAG.

The base container logs to the UI with S3 remote logging with the KubernetesExecutor using an approach similar to this [article](https://medium.com/localize-city/handling-airflow-logs-with-kubernetes-executor-25c11ea831e4).

I'd like to get the logs from the git-sync sidecar container into the UI for ease of troubleshooting in case the DAG doesn't start, git issues, etc. How would I go about doing that? I've pondered the idea of getting some custom functions into the git-sync container to get it to push the logs up, but that would seem to require getting all of the Airflow libraries on there, which seems quite convoluted, since the container itself is supposed to be lightweight, consisting of just what it needs for git-sync. 
I've also thought about somehow having the base container pull the logs from the git-sync container put it in S3 for remote logging, but not sure if that's possible without some massive overhaul.

Any recommended approaches for going about this?

**Comment by potiuk** [comment, marked as answer]
UI only show task containers. You need to have separate remote logigng for your cluster to get what you want. You can configure your K8S cluster to send logs to the same solution you use (cloudwatch/elasticsearch etc. in the way that is native to the k8s cluster you use). Airlfow UI shows only task output nothing else and that's by design, you need ot chose logging solution for the whole deployment of yours independently of Airflow.

This is the recommended solution.
